src/libAtomVM/utils.h

```diff
@@ -102,23 +102,53 @@
         #define ENDIAN_SWAP_32(value) ((((value) & 0xFF) << 24) | (((value) & 0xFF00) << 8) | (((value) & 0xFF0000) >> 8) | (((value) & 0xFF000000) >> 24))
     #endif
 
-#else
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+
+    #define READ_64_UNALIGNED(ptr) \
+        ( (((uint64_t) ((uint8_t *)(ptr))[0]) << 56) | (((uint64_t) ((uint8_t *) (ptr))[1]) << 48) | \
+            (((uint64_t) ((uint8_t *)(ptr))[2]) << 40) | (((uint64_t) ((uint8_t *) (ptr))[3]) << 32) | \
+            (((uint64_t) ((uint8_t *)(ptr))[4]) << 24) | (((uint64_t) ((uint8_t *) (ptr))[5]) << 16) | \
+            (((uint64_t) ((uint8_t *)(ptr))[6]) << 8) | (((uint64_t) ((uint8_t *) (ptr))[7])) )
+
+    #define WRITE_64_UNALIGNED(ptr, val) \
+        { \
+            ((uint8_t *)(ptr))[0] = (((uint64_t) val) >> 56) & 0xff; \
+            ((uint8_t *)(ptr))[1] = (((uint64_t) val) >> 48) & 0xff; \
+            ((uint8_t *)(ptr))[2] = (((uint64_t) val) >> 40) & 0xff; \
+            ((uint8_t *)(ptr))[3] = (((uint64_t) val) >> 32) & 0xff; \
+            ((uint8_t *)(ptr))[4] = (((uint64_t) val) >> 24) & 0xff; \
+            ((uint8_t *)(ptr))[5] = (((uint64_t) val) >> 16) & 0xff; \
+            ((uint8_t *)(ptr))[6] = (((uint64_t) val) >> 8) & 0xff; \
+            ((uint8_t *)(ptr))[7] = ((uint64_t) val) & 0xff; \
+        }
+
     #define READ_32_ALIGNED(ptr) \
         (*((uint32_t *) (ptr)))
 
     #define READ_32_UNALIGNED(ptr) \
```

Comment:
I like the new indentation and alignment so much it would be good to see it here too.

Reply:
do you mean `\` identation? Keep in mind this is just a copy and paste of what we have some lines above.

Reply:
No just the line-per-operation pattern.  Less concerned about the `\`